api/cmd/etcd-launcher/main.go

```diff
@@ -0,0 +1,139 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+	"syscall"
+)
+
+const (
+	initialStateEnvName      = "INITIAL_STATE"
+	initialClusterEnvName    = "INITIAL_CLUSTER"
+	defaultClusterSize       = 3
+	defaultEtcdctlAPIVersion = "3"
+)
+
+type envConfig struct {
+	namespace             string
+	clusterSize           int
+	podName               string
+	podIP                 string
+	etcdctlAPIVersion     string
+	dataDir               string
+	token                 string
+	enableCorruptionCheck bool
+}
+
+func main() {
+	// normal workflow, we don't do migrations anymore
+	config, err := getConfigFromEnv()
+	if err != nil {
+		log.Fatalf("failed to get launcher configuration: %v", err)
+	}
+
+	// not required, will leave it for now.
+	os.Setenv(initialStateEnvName, "new")
+	os.Setenv(initialClusterEnvName, initialMemberList(config.clusterSize, config.namespace))
+
+	log.Print("initializing etcd..")
+	log.Printf("initial-state: %s", os.Getenv(initialStateEnvName))
+	log.Printf("initial-cluster: %s", os.Getenv(initialClusterEnvName))
+
+	err = syscall.Exec(
```

Comment:
why not os/exec?

Reply:
Because I don't want to manage the process, I want to replace it with the command. Original script did the same trick with bash.